lib/fast_jsonapi/serialization_core.rb

```diff
@@ -73,13 +74,21 @@ def links_hash(record, params = {})
       end
 
       def attributes_hash(record, params = {})
-        attributes_to_serialize.each_with_object({}) do |(key, method), attr_hash|
+        attributes = attributes_to_serialize.each_with_object({}) do |(key, method), attr_hash|
           attr_hash[key] = if method.is_a?(Proc)
             method.arity == 1 ? method.call(record) : method.call(record, params)
           else
             record.public_send(method)
           end
         end
+
+        self.optional_attributes_to_serialize = {} if self.optional_attributes_to_serialize.nil?
+        optional_attributes_to_serialize.each do |key, details|
+          method_name, check_proc = details
+          attributes[key] = record.send(method_name) if check_proc.call(record)
```

Comment:
@andyjeffries couldn't this be `check_proc.call(record, params)` to get access to the custom parameters? That'd be necessary for our implementation, and if you're checking admin key I'd imagine you'd need that as well. I feel like most context-driven responses won't rely on the record's data, but outside data given in params.

Reply:
Yep, that's a nice touch too.